driver/optcompile.ml

```diff
@@ -70,31 +89,106 @@ let emit unix i =
     (Unit_info.prefix i.target)
     ~progname:(Unit_info.source_file i.target)
 
-let implementation unix ~(flambda2 : flambda2) ~start_from ~source_file
-    ~output_prefix ~keep_symbol_tables =
-  let backend info ({ structure; coercion; _ } : Typedtree.implementation) =
+type starting_point =
```

Comment:
Since this is not really "starting_point" (`-instantiate` is incompatible with `-stop-after`), let's get a new name. How about:
```
type implementation_mode =
  | Normal
  | Emit
  | Instantiate of ..
```

Reply:
It's exactly a starting point, though? It's the allowed values for the `~start_from` parameter to `implementation`, plus one more place we can start from. (Actually, I have no idea why `-instantiate` is deemed incompatible with `-stop-after`. I'll take that out.)

Reply:
(It's not in fact incompatible, it was just a string that said so)

Reply:
What bothers me is that the name `type starting_point` sounds like just an alias for `Clflags.Compiler_pass.t` (in the sense that it specifies a point in the compiler pass), but in fact it also contains `Instantiate` which is not a point in the compiler pass, but a mode.

Reply:
It's meant to be a generalization: sometimes we start from a compiler pass but sometimes we don't. (It might make sense to have one of the constructors be `Pass of Clflags.Compiler_pass.t` but I wanted to keep the constructors the same to keep as much of the `match` the same as upstream.)